lua/lazyvim/plugins/extras/lang/lean.lua

```diff
@@ -0,0 +1,141 @@
+          -- course may also use a snippet engine directly to do
+          -- this if so desired.
+          wknight = "♘",
+        },
+        -- Change if you don't like the backslash
+        -- (comma is a popular choice on French keyboards)
+        leader = "\\",
+      },
+
+      -- Enable suggested mappings?
+      --
+      -- false by default, true to enable
+      mappings = true,
+
+      -- Infoview support
+      infoview = {
+        -- Automatically open an infoview on entering a Lean buffer?
+        -- Should be a function that will be called anytime a new Lean file
+        -- is opened. Return true to open an infoview, otherwise false.
+        -- Setting this to `true` is the same as `function() return true end`,
+        -- i.e. autoopen for any Lean file, or setting it to `false` is the
+        -- same as `function() return false end`, i.e. never autoopen.
+        autoopen = true,
+
+        -- Set infoview windows' starting dimensions.
+        -- Windows are opened horizontally or vertically depending on spacing.
+        width = 50,
+        height = 20,
+
+        -- Put the infoview on the top or bottom when horizontal?
+        -- top | bottom
+        horizontal_position = "bottom",
+
+        -- Always open the infoview window in a separate tabpage.
+        -- Might be useful if you are using a screen reader and don't want too
+        -- many dynamic updates in the terminal at the same time.
+        -- Note that `height` and `width` will be ignored in this case.
+        separate_tab = false,
+
+        -- Show indicators for pin locations when entering an infoview window?
+        -- always | never | auto (= only when there are multiple pins)
+        indicators = "auto",
+      },
+
+      -- Progress bar support
+      progress_bars = {
+        -- Enable the progress bars?
+        enable = true,
+        -- What character should be used for the bars?
+        character = "│",
+        -- Use a different priority for the signs
+        priority = 10,
+      },
+
+      -- Redirect Lean's stderr messages somehwere (to a buffer by default)
+      stderr = {
+        enable = true,
+        -- height of the window
+        height = 5,
+        -- a callback which will be called with (multi-line) stderr output
+        -- e.g., use:
+        --   on_lines = function(lines) vim.notify(lines) end
+        -- if you want to redirect stderr to `vim.notify`.
+        -- The default implementation will redirect to a dedicated stderr
+        -- window.
+        on_lines = nil,
+      },
+    },
+  },
+  {
+    -- Lean4 has it's own built in lsp
+    -- this is automatically added by lean.nvim
+    "neovim/nvim-lspconfig",
+  },
+  {
+    "hrsh7th/nvim-cmp",
+    optional = true,
+    opts = function(_, opts)
+      opts.sources = opts.sources or {}
+      table.insert(opts.sources, { { name = "nvim_lsp" }, { name = "path" }, { name = "buffer" } })
+    end,
+  },
```

Comment:
I don't think these are needed, since `nvim-lspconfig` is just an empty spec and `nvim-cmp` has already these values by default in LazyVim?
```suggestion
 
```